docs/source/tutorial/train_loop.rst

```diff
@@ -0,0 +1,328 @@
+How to write a training loop in Chainer
+```````````````````````````````````````
+
+.. currentmodule:: chainer
+
+In this tutorial section, we will learn how to train a deep neural network to classify images of hand-written digits in the popular MNIST dataset. This dataset contains 50,000 training examples and 10,000 test examples. Each example is a set of a 28 x 28 greyscale image and a corresponding class label. Since the digits from 0 to 9 are used, there are 10 classes for the labels.
+
+Chainer provides a feature called :class:`~chainer.training.Trainer` that can simplify the training procedure of your model. However, it is also good to know how the training works in Chainer before starting to use the useful :class:`~chainer.training.Trainer` class that hides the actual processes. For advanced users, it sometimes needs to write their own custom training loop.
```

Comment:
For advanced users, it sometimes needs to write their own custom training loop.
->
Writing your own training loop can be useful for learning how `Trainer` works or for implementing features not included in the standard trainer.